framework/src/tinyalsa/tinyalsa.c

```diff
@@ -439,7 +441,7 @@ static int pcm_mmap_transfer_areas(struct pcm *pcm, char *buf, unsigned int offs
 	while (size > 0) {
 		frames = size;
 		pcm_mmap_begin(pcm, &pcm_areas, &pcm_offset, &frames);
-		pcm_areas_copy(pcm, pcm_offset, buf, offset, frames);
+		pcm_areas_copy(pcm, pcm_offset, pcm_areas, buf, offset, frames);
```

Comment:
I have a doubt here, shouldn't min(frames,size) be passed as argument to pcm_areas_copy , because if size is less than frames, then we can end up overwriting memory which is not allocated to this buffer.